src/types.ts

```diff
@@ -231,13 +231,21 @@ interface PhaseMap<G extends any = any, CtxWithPlugins extends Ctx = Ctx> {
   [phaseName: string]: PhaseConfig<G, CtxWithPlugins>;
 }
 
-export interface Game<G extends any = any, CtxWithPlugins extends Ctx = Ctx> {
+export interface Game<
+  G extends any = any,
+  CtxWithPlugins extends Ctx = Ctx,
+  SetupData extends any = any
+> {
   name?: string;
   minPlayers?: number;
   maxPlayers?: number;
   disableUndo?: boolean;
   seed?: string | number;
-  setup?: (ctx: CtxWithPlugins, setupData?: any) => any;
+  setup?: (ctx: CtxWithPlugins, setupData?: SetupData) => any;
+  validateSetupData?: (
+    setupData: SetupData | undefined,
+    numPlayers: number
+  ) => string | undefined;
```

Comment:
This return type seems different from what might be expected from the implementation, which expects `string | null`. I think `string | undefined` might actually be better, because that allows implicit returns if all is well, rather than requiring the function to return `null`. What do you think?

Reply:
The explicit distinction between `null` and `undefined` in TypeScript still trips me up every time as years of Javascript development have made me treat them as essentially equivalent. I updated the documentation, the server check, and tests to explicitly use `undefined`.